Steamfitter.Api/Services/TaskService.cs

```diff
@@ -244,6 +245,39 @@ public async STT.Task<IEnumerable<SAVM.Result>> ExecuteAsync(Guid id, Cancellati
             return  new List<SAVM.Result>();
         }
 
+        public async STT.Task<Guid?> ExecuteForGradeAsync(Dictionary<string, string> taskSubstitutions, CancellationToken ct)
+        {
+            var userId = _user.GetId();
+            var scenarioId = Guid.Parse(taskSubstitutions["scenarioid"]);
```

Comment:
any reason this can't be an explicit parameter?

Reply:
The thought here is that there may be more than one file in this task tree that requires content substitution.  The private method MakeTaskSubstitutionsAsync loops through the Dictionary to see if there is a content substitution for each File Upload task.  The Dictionary allows for any number of files to get new content.  I thought that this might also be useful in the infinity challenges.  The actual answer file could be uploaded allowing for a unique task for each execution attempt.

Reply:
I didn't mean remove the Dictionary entirely, but couldn't ScenarioId be an explicit parameter in addition to the Dictionary since you know that's always required?

Reply:
That makes sense.  I also made StartTaskName explicit, as well.  Thank you!